### PR TITLE
Tests and stuff

### PR DESCRIPTION
**Pull request by 17Halbe**
Hi there, nice to meet you here as well.

I did a lot of stuff besides adding the tests. I really hope that's ok for you, otherwise just edit it out again! ;)
I changed the variable names since (see gitter) they really confused me. :joy: 
I changed them to `REPORT_MAIL` and `REPORT_INTERVAL`. That way, I thought, people might recognise them as more "affiliated" to each other. And they(the people) probably just don't need to know if that's postfix, dovecot, whatever related.. What do you think? 
Furthermore I gave `sendmail` another try and after some time got it working. So we can ditch that mailx stuff.
That's about it. Found some bugs too! Yeah!! :grin: 
Really good work! Thank you!!

Ah, yes, I forgot to update the documentation with the new variable names..

### CODE REVIEW

File: target/start-mailserver.sh [hunk 1]
```diff
@@ -1079,6 +1083,32 @@ function _setup_elk_forwarder() {
...
+	LOGROTATE="/var/log/mail/mail.log\n{\n  compress\n  delaycompress\n"
+
+	case "$REPORT_INTERVAL" in
+		"daily" )
+			notify 'inf' "Setting postfix summary interval to daily"
+			LOGROTATE="$LOGROTATE  rotate 7\n  daily\n"
+			;;
+		"weekly" )
+			notify 'inf' "Setting postfix summary interval to weekly"
+			postconf -e "$(postconf | grep '^mynetworks =') 172.16.0.0/12"
```
Comment (17Halbe) [inline target/start-mailserver.sh]: Why are the postconf commands here in weekly and not in daily and then different again in monthly?

File: target/start-mailserver.sh [hunk 2]
```diff
@@ -1079,6 +1083,32 @@ function _setup_elk_forwarder() {
...
+			LOGROTATE="$LOGROTATE  rotate 4\n  weekly\n"
+			;;
+		"monthly" )
+			notify 'inf' "Setting postfix summary interval to monthly"
+			postconf -e "$(postconf | grep '^mynetworks =') $container_ip/32"
```
Comment (17Halbe) [inline target/start-mailserver.sh]: Here is the other one

**Comment by 17Halbe** [comment]
Is `sharedscript` neccessary in the logrotate config? (I probably removed it a bit prematurely) :/
I added the `copytruncate` option to the logrotate config, it wasn't creating a new logfile on my machine..

File: target/start-mailserver.sh (hunk 1 above)
Comment (akmet) [reply target/start-mailserver.sh]: oh shit. This should never been here. My fault. I copied the case part and it seems I either did not saved the corrected files when submitting the commit or I missed something else. I'm investigating!

**Comment by akmet** [comment]
As I mentioned in gitter: I really welcome your edits!
I think `REPORT_MAIL `is choosen good, but won't be `LOGROTATE_INTERVAL `more specific? Because `REPORT_ `seems to be a reliance to the postfix log summary (called report). But it should be more seen as an independent function / feature. Maybe we could adapt the variable for other logs too?
Nice to see that you got sendmail working, will have a look at it maybe tomorrow..

In logrotate you have two ways for excecuting a shell script after the last logrotate.
- use lastrotate or
- use postrotate with sharedscript enabled

since sharedscript will also "call" create and that will first create the logfile after rotation before excecuting the script. I thought this would create the new (empty) log file and after that processes with calling the shell script mentioned in the conf. Since you had problems with automatical creating the new logfile I maybe just misunderstood the man page? When there is a more elegant way to implement the requested behaviour then we always should choose that.

File: target/start-mailserver.sh [hunk 3]
```diff
@@ -136,8 +136,9 @@ function register_functions() {
 		_register_setup_function "_setup_postfix_virtual_transport"
 	fi
 
-  _register_setup_function "_setup_postfix_summary"
-  _register_setup_function "_setup_environment"
+  if [ "$REPORT_MAIL" != 0 ]; then
```
Comment (akmet) [inline target/start-mailserver.sh]: That will result in setting up logrotate for mail.log only if the user specified an email at REPORT_MAIL

File: target/start-mailserver.sh [hunk 4]
```diff
@@ -136,8 +136,9 @@ function register_functions() {
 		_register_setup_function "_setup_postfix_virtual_transport"
 	fi
 
-  _register_setup_function "_setup_postfix_summary"
-  _register_setup_function "_setup_environment"
```
Comment (akmet) [inline target/start-mailserver.sh]: I think we - maybe - should keep `_setup_environment` ... :-D

Comment (17Halbe) [reply target/start-mailserver.sh]: :joy: You think so? :innocent:

**Comment by 17Halbe** [comment]
Than we misunderstood each other. That was actually my initial question on gitter. So we do have two different independent variables. And to make it clearer, logrotate should be called EVERY time, and REPORT only when the user set up the report email, right?
Yes, I reread the man page and it says, that it's automatically creating a new file. Maybe the test script was too fast and the file wasn't created yet. `copytruncate` should be safer though, since it only deletes the content of the file and not the file itself.

**Pull request closed without merging**